packages/fetch-and-compile/test/fixture.js

```diff
@@ -0,0 +1,2825 @@
+const mainnetData = {
```

Comment:
These test fixtures are huge!  Wouldn't it make more sense to use something smaller?

Reply:
Yeah it would be nice to have something smaller. I think we just used the actual response from the call it makes in the test.

Reply:
Actually, on looking at this some more, what bugs me here is not so much the *size* but rather the lack of *structure*.  I think this would be a lot more swallowable if, instead of using double-quoted string literals with `\n`s everywhere for line breaks, it used backticked-string literals with actual linebreaks.  (We do this a bunch in the debugger tests for including Solidity sources.)  Likely, you should make that change even if you change it to something smaller.

Reply:
I was actually able to decrease the size of the fixture without failing the tests. It seems like just the first line of the contract and the contract name will suffice to pass the tests.

Reply:
Yeah I wouldn't recommend going that far!  We do want an actual fully-working fixture; we want to be able to add more tests in the future.  Please don't optimize the fixture to only pass the particular tests that you happened to add!

Reply:
Oh wait I just realize I misunderstood what you meant, you just meant stripping the contract down to the bare minimum, not going even further than that; for some reason I misunderstood and thought you meant the latter.  Yeah that would be fine, although I might put in at least some token function that can be checked for in the future or something if we want to expand the tests.

Reply:
So, uh, @leeftk, any action on this?  What's going on here?

Reply:
I spoke with @amal and he suggested removing the `\n` and the '+' as you also did above. So I went that route instead. I didn't edit the ABI nor the Source text. That's what came back from the response. The only thing I changed was removing the line breaks as I figured it leave the data as is would give us the option to expand our tests in the future if need be. But if we want to go with the a different option then I can remove everything besides the first line of each Source code and it will still compile. But I can't leave just a function in there as you suggested above as it won't compile and the tests fail. So besides what I submitted the only other option is leaving just the `Pragma solidity <version>` line.

Reply:
> I spoke with @amal and he suggested removing the \n and the '+' as you also did above. So I went that route instead.

I don't understand.  There's no concatenation, sure, but I'm still seeing a lot of `\r\n` everywhere.  I suggested using template strings (backticks) so that you can use actual linebreaks, so that it looks more structured.  (You could do the same with the ABI field, so that it's not full of `\"`s.  Is that something you plan to do?

> I didn't edit the ABI nor the Source text. That's what came back from the response.

Oh huh!  My misunderstanding.  They looked mismatched to me.  But I guess not?

> But if we want to go with the a different option then I can remove everything besides the first line of each Source code and it will still compile.

No, please do not do that!

OK, let me explain in more detail what I think are good vs bad ways to strip down the tests.

Fundamentally, the tests should be *coherent*.  They don't have to be a real contract instance off of Etherscan, but they should be what the output *would* be if you wrote a real contract, deployed it, and then verified it on Etherscan.  Remember: We want to be able to write more detailed tests later!

When I talked about stripping down the fixtures, I didn't mean just deleting parts of it without regard for the rest.  I meant rewriting the whole fixture as a new, minimal contract, with all fields adjusted appropriately.

At the very least, the source text ought to include an actual contract, not just a pragma line!  This whole package is based around the idea of *contracts* with verified sources, not just verified sources by themselves.  Note, by the way, that removing the contract entirely is incompatible with the suggestion I made about how the contract name should be tested.

It seems there was a bit of a misunderstanding here -- you talked about stripping down the source file:
> I was actually able to decrease the size of the fixture without failing the tests. It seems like just the first line of the contract and the contract name will suffice to pass the tests.
and I was like, no, don't do that!  But then on rereading I concluded you meant the first line of the *contract*, i.e. the contract declaration, rather than the first line of the *source file*.  After all, you were checking the contract name in the tests, so you must have included the contract, right?  And a minimal contract declararation like that seemed fine, although I suggested including a function too.

(Btw, you say you got compile errors... what compile errors?  Could you go into more detail there?  I ask because when you say you "left in a function", I'm not clear on what exactly you did.  What I had meant was to just put in like a little stub function that does nothing, but you may have taken a different approach...)

But I hadn't yet noticed that, in your tests, you were getting the contract name out of `sourceInfo` rather than `compilations` -- which would allow you to get the contract name *without* having an actual contract.  And now with this comment you've made it explicit that, no, you really were suggesting stripping out the entire contract.

So, please don't do that.  There should, in fact, be a contract.  It can be a minimal one, but it must be a contract.  Moreover, if you go this route, the rest of the response has to be adjusted to match; you can't just alter one component of it and leave the rest the same.

That said, as I've pointed out above, if you just switch to using backticked literals (template strings), so that the whole thing looks more structured, there may just not be much of a need to strip it down in the first place.

Another approach, which you might want to consider, would be to pull the source text out into its own file, `Whatever.sol`, and read that text in here, rather than having it inline, if it's so long.  You could also possibly have `Whatever.abi.json`, etc.  Basically, breaking it up might make it easier to process.  Then having it be so large would be less of a concern.

I'm hoping that makes it clear what I'm asking for?  As always, if it's not, please ask me about any parts that are unclear.

Reply:
Awesome thanks for the clarity in this comment. So basically I decided to go with your last suggestion of splitting up the `fixture.js` file into a few different files to make it a bit neater and also easier to consume. I created individual files like `arbitrum.abi.js` for the abi's  and did the same for the source info (labeled `arbitrum.sol.js` in this example). Before pushing this up I just wanted to confirm that this is what you were getting at in that last part where you mentioned. 

> Another approach, which you might want to consider, would be to pull the source text out into its own file, Whatever.sol, and read that text in here, rather than having it inline, if it's so long. You could also possibly have Whatever.abi.json, etc. Basically, breaking it up might make it easier to process. Then having it be so large would be less of a concern.

If so then we should be good!

Reply:
> I created individual files like arbitrum.abi.js for the abi's and did the same for the source info (labeled arbitrum.sol.js in this example). Before pushing this up I just wanted to confirm that this is what you were getting at in that last part where you mentioned.

Well, partly, but also if you are splitting them up like that they should probably be the appropriate types of files -- like, `.sol` and `.json` -- rather than forced into being JS files.

I'm going to go ahead and take a guess... did you make them JS files so that you could read them with `require`?  I'd suggest just using the appropriate file types and then reading those the appropriate way (like, using `fs.readFileSync`, and then for the JSON potentially doing `JSON.parse` if that's appropriate in this context).

Reply:
I'm going to resolve this but if we need to try a different method just let me know.

Reply:
Yes, I think I'm going to have to insist on doing this properly.  Please use the appropriate file types and read them in as appropriate; don't stuff them in `.js` files and use `require` like this.

(Moreover, if you *were* to put them in `.js` files, you should -- as I've mentioned above -- be using backticks, i.e. template literals, to make them more readable!  But in actuality these shouldn't be in `.js` files in the first place.)

Reply:
So when we fetch a contract from Etherscan you're saying that in the response in `SourceCode` that's actually Solidity and not JS? I was under the assumption that the response was a JS object which is why I put them in a JS file.

Reply:
Your question seems to indicate a confusion of levels.  I could just say "Yes, that's Solidity", but I'm not sure that would properly address your confusion.

(Also, remember here, I'm not *just* talking about the Solidity sources, which ought to be in `.sol` files, but also the ABI JSONs, which ought to be in `.json` files, and treated as JSON, not as Javascript.  JSON and Javascript may often be identical in representation, but they are conceptually different and should not be treated as interchangeable!)

...I originally started writing a much longer explanation, but I couldn't get it into a form I was happy with, so I'm just going to give a really short answer and hope it suffices: The overall response that we get back from Etherscan is JSON.  The `SourceCode` field of that JSON is a string, which is represented in the response as a JSON-encoded string.  The contents of the underlying string, that is represented by this JSON-encoded form, are the Solidity source.

(Remember: Everything that gets sent over the wire is ultimately just a bunch of bytes.  It's up to us to impose an appropriate interpretation on it!)

This test fixture you are creating, meanwhile, is not JSON, but rather is a Javascript object that will be serialized to JSON.  So, let's ignore the JSON layer, since here we are not actually dealing with JSON.  Then, looking at this Javascript object, it has a `SourceCode` field, which is a string, and that string consists of the Solidity source.  So, the `SourceCode` field of the object you are setting up, is a member of a Javascript object, and as such is a Javascript string, and the contents of that string are Solidity source.

You want to have that string, the Solidity source, represented by a file.  The appropriate way to represent a string by a file is for the file's contents to consist of the contents of that string.  (And since the contents of that string are Solidity source, it would be appropriate to give that file a name ending in `.sol`.)  To instead represent a string in a file by making that file a JS source file, and encoding the string as an JS string literal inside that JS source file, is needless complication.

...I'm not sure if that is sufficient to clear up your confusion; I realize that this is perhaps not the most in-depth, and I don't know where you're starting from.  Well -- if you want to go deeper into questions of data representation, I think Slack or Zoom might be a better place to ask me about them rather than here... we can go over examples or whatever if you want!

Reply:
Okay thanks for the explanation here. I paired with Tyler and he went into a bit more detail as well as to why this is important. I wasn't understanding at first from the comments but after getting a visual representation I was able to understand what you meant here so thanks for the detailed response, it really helped. I created `.sol` files for the Solidity files and also changed the ABI's to `.json` files. Also to answer an earlier question that you asked, I had trouble finding smaller contracts and it ended up taking more time than I expected. The only smaller contract I was able to find was one for the Arbitrum network so I switched that one out and left the others as is. I guess most contracts these days use the OpenZepplin libraries so even if the main contract is small it comes with a bunch of extra stuff that I don't want to have to deal with. So I chose to just use the contracts I had originally and go with one of the of methods you suggested of how to decrease the fixture. Hope that helps cover everything. Thanks Harry!